File: src/algebra1_graphing_lines_easy.py
"""
algebra1 - graphing_lines (easy)
Generated: 2026-02-22T03:54:07.817087
"""

from problem_utils import *
from svg_utils import Graph

def generate():
    problem_type = randint(1, 5)
    
    if problem_type == 1:
        # Find slope from two points (easiest)
        x1, y1 = randint(-5, 5), randint(-5, 5)
        x2, y2 = x1 + randint(1, 4), y1 + randint(-4, 4)
        while y2 == y1:
            y2 = y1 + randint(-4, 4)
        
        slope = Rational(y2 - y1, x2 - x1)
        
        g = Graph(x_range=(-8, 8), y_range=(-8, 8), width=300, height=300)
        g.point(x1, y1, f"({x1},{y1})")
        g.point(x2, y2, f"({x2},{y2})")
        line_expr = slope * (x - x1) + y1
        g.plot(line_expr)
        
        return problem(
            question=f"Find the slope of the line passing through the points $({x1}, {y1})$ and $({x2}, {y2})$.",
            answer=slope,
            difficulty=(1000, 1100),
            topic="algebra1/graphing_lines",
            solution=steps(
                f"Use the slope formula: $m = \\frac{{y_2 - y_1}}{{x_2 - x_1}}$",
                f"$m = \\frac{{{y2} - ({y1})}}{{{x2} - ({x1})}} = \\frac{{{y2 - y1}}}{{{x2 - x1}}}$",
                f"$m = {latex(slope)}$"
            ),
            image=g.render()
        )
    
    elif problem_type == 2:
        # Find y-intercept from graph or equation in slope-intercept form
        m = nonzero(-3, 3)
        b_val = randint(-6, 6)
        
        equation = Eq(y, m*x + b_val)
        
        return problem(
            question=f"What is the $y$-intercept of the line ${latex(equation)}$?",
            answer=b_val,
            difficulty=(1050, 1150),
            topic="algebra1/graphing_lines",
            solution=steps(
                f"The equation is in slope-intercept form: $y = mx + b$",
                f"The $y$-intercept is the constant term $b$",
                f"$b = {b_val}$"
            )
        )
    
    elif problem_type == 3:
        # Identify slope from slope-intercept form
        m = Rational(nonzero(-4, 4), choice([1, 2, 3, 4]))
        b_val = randint(-5, 5)
        
        equation = Eq(y, m*x + b_val)
        
        return problem(
            question=f"What is the slope of the line ${latex(equation)}$?",
            answer=m,
            difficulty=(1000, 1100),
            topic="algebra1/graphing_lines",
            solution=steps(
                f"The equation is in slope-intercept form: $y = mx + b$",
                f"The slope is the coefficient of $x$",
                f"$m = {latex(m)}$"
            )
        )
    
    elif problem_type == 4:
        # Find x-intercept (where y=0)
        m = nonzero(-3, 3)
        b_val = nonzero(-6, 6)
        
        # x-intercept when y=0: 0 = mx + b, so x = -b/m
        x_intercept = Rational(-b_val, m)
        
        equation = Eq(y, m*x + b_val)
        
        return problem(
            question=f"Find the $x$-intercept of the line ${latex(equation)}$.",
            answer=x_intercept,
            difficulty=(1150, 1250),
            topic="algebra1/graphing_lines",
            solution=steps(
                f"The $x$-intercept occurs when $y = 0$",
                f"$0 = {m}x + {b_val}$",
                f"${-b_val} = {m}x$",
                f"$x = {latex(x_intercept)}$"
            )
        )
    
    else:
        # Graph a simple line (identify slope and y-intercept, then plot)
        m = choice([-2, -1, 1, 2])
        b_val = randint(-3, 3)
        
        g = Graph(x_range=(-5, 5), y_range=(-5, 5), width=300, height=300)
        g.plot(m*x + b_val)
        g.point(0, b_val, f"(0,{b_val})")
        
        equation = Eq(y, m*x + b_val)
        
        return problem(
            question=f"The graph shows the line ${latex(equation)}$. What point does the line cross the $y$-axis?",
            answer=fmt_tuple(0, b_val),
            difficulty=(1100, 1200),
            topic="algebra1/graphing_lines",
            solution=steps(
                f"The $y$-intercept is where the line crosses the $y$-axis (where $x = 0$)",
                f"From the equation $y = {m}x + {b_val}$, when $x = 0$:",
                f"$y = {m}(0) + {b_val} = {b_val}$",
                f"The point is $(0, {b_val})$"
            ),
            image=g.render()
        )

emit(generate())

File: src/algebra1_graphing_lines_easy.jl
# algebra1 - graphing_lines (easy)
# Generated: 2026-03-08T19:56:27.899085

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x y begin
    set_topic!("algebra1/graphing_lines")
    
    problem_type = choice([
        :slope_from_two_points,
        :y_intercept_from_equation,
        :point_on_line,
        :slope_intercept_form,
        :x_intercept,
        :parallel_slope,
        :perpendicular_slope,
        :equation_from_point_slope,
        :slope_from_graph_description,
        :find_y_given_x
    ])
    
    if problem_type == :slope_from_two_points
        x1, y1 = randint(-12, 12), randint(-12, 12)
        x2, y2 = randint(-12, 12), randint(-12, 12)
        while x1 == x2
            x2 = randint(-12, 12)
        end
        m = (y2 - y1) // (x2 - x1)
        
        problem(
            question="Find the slope of the line passing through the points \\(($(x1), $(y1))\\) and \\(($(x2), $(y2))\\).",
            answer=m,
            difficulty=(700, 900),
            solution=steps(
                "Use the slope formula: \$m = \\frac{y_2 - y_1}{x_2 - x_1}\$",
                sol("Substitute", "m = \\frac{$(y2) - ($(y1))}{$(x2) - ($(x1))} = \\frac{$(y2 - y1)}{$(x2 - x1)}"),
                sol("Answer", m)
            ),
            time=90
        )
        
    elseif problem_type == :y_intercept_from_equation
        m = nonzero(-8, 8)
        b = randint(-15, 15)
        
        problem(
            question="What is the y-intercept of the line \$y = $(m)x + $(b)\$?",
            answer=b,
            difficulty=(700, 850),
            solution=steps(
                "The equation is in slope-intercept form: \$y = mx + b\$",
                "The y-intercept is the constant term \$b\$",
                sol("Answer", b)
            ),
            time=60
        )
        
    elseif problem_type == :point_on_line
        m = nonzero(-6, 6)
        b = randint(-10, 10)
        x_val = randint(-8, 8)
        y_val = m * x_val + b
        
        problem(
            question="Does the point \\(($(x_val), $(y_val))\\) lie on the line \$y = $(m)x + $(b)\$? Answer 'yes' or 'no'.",
            answer="yes",
            answer_type="text",
            difficulty=(750, 950),
            solution=steps(
                sol("Substitute \$x = $(x_val)\$ into the equation", "y = $(m)($(x_val)) + $(b)"),
                sol("Calculate", "y = $(m * x_val) + $(b) = $(y_val)"),
                "Since this matches the y-coordinate, the answer is yes"
            ),
            time=75
        )
        
    elseif problem_type == :slope_intercept_form
        m = nonzero(-7, 7)
        x_pt = randint(-10, 10)
        y_pt = randint(-10, 10)
        b = y_pt - m * x_pt
        
        problem(
            question="Write the equation of the line with slope \$m = $(m)\$ passing through \\(($(x_pt), $(y_pt))\\) in slope-intercept form.",
            answer="y = $(m)x + $(b)",
            answer_type="text",
            difficulty=(900, 1100),
            solution=steps(
                "Use point-slope form: \$y - y_1 = m(x - x_1)\$",
                sol("Substitute", "y - $(y_pt) = $(m)(x - $(x_pt))"),
                sol("Expand and simplify", "y = $(m)x + $(b)")
            ),
            time=120
        )
        
    elseif problem_type == :x_intercept
        m = nonzero(-8, 8)
        b = nonzero(-15, 15)
        x_int = -b // m
        
        problem(
            question="Find the x-intercept of the line \$y = $(m)x + $(b)\$.",
            answer=x_int,
            difficulty=(800, 1000),
            solution=steps(
                "Set \$y = 0\$ and solve for \$x\$",
                sol("Equation", "0 = $(m)x + $(b)"),
                sol("Solve", "x = $(x_int)")
            ),
            time=90
        )
        
    elseif problem_type == :parallel_slope
        m = nonzero(-9, 9)
        b = randint(-12, 12)
        
        problem(
            question="What is the slope of any line parallel to \$y = $(m)x + $(b)\$?",
            answer=m,
            difficulty=(750, 950),
            solution=steps(
                "Parallel lines have equal slopes",
                "The slope of the given line is \$m = $(m)\$",
                sol("Answer", m)
            ),
            time=60
        )
        
    elseif problem_type == :perpendicular_slope
        m = nonzero(-8, 8)
        b = randint(-12, 12)
        m_perp = -1 // m
        
        problem(
            question="What is the slope of any line perpendicular to \$y = $(m)x + $(b)\$?",
            answer=m_perp,
            difficulty=(850, 1050),
            solution=steps(
                "Perpendicular lines have slopes that are negative reciprocals",
                "The slope of the given line is \$m = $(m)\$",
                sol("Answer", "m_{\\perp} = -\\frac{1}{$(m)} = $(m_perp)")
            ),
            time=90
        )
        
    elseif problem_type == :equation_from_point_slope
        x1, y1 = randint(-10, 10), randint(-10, 10)
        m = nonzero(-6, 6)
        b = y1 - m * x1
        
        problem(
            question="Find the equation of the line passing through \\(($(x1), $(y1))\\) with slope \$m = $(m)\$. Write your answer in the form \$y = mx + b\$.",
            answer="y = $(m)x + $(b)",
            answer_type="text",
            difficulty=(900, 1150),
            solution=steps(
                sol("Use point-slope form", "y - $(y1) = $(m)(x - $(x1))"),
                "Expand the right side and add \$$(y1)\$ to both sides",
                sol("Slope-intercept form", "y = $(m)x + $(b)")
            ),
            time=120
        )
        
    elseif problem_type == :slope_from_graph_description
        rise = nonzero(-10, 10)
        run = nonzero(1, 10)
        m = rise // run
        
        problem(
            question="A line rises $(rise) units for every $(run) units of horizontal movement. What is its slope?",
            answer=m,
            difficulty=(700, 900),
            solution=steps(
                "Slope is the ratio of vertical change to horizontal change",
                sol("Formula", "m = \\frac{\\text{rise}}{\\text{run}}"),
                sol("Answer", "m = \\frac{$(rise)}{$(run)} = $(m)")
            ),
            time=75
        )
        
    else  # :find_y_given_x
        m = nonzero(-7, 7)
        b = randint(-12, 12)
        x_val = randint(-10, 10)
        y_val = m * x_val + b
        
        problem(
            question="If \$y = $(m)x + $(b)\$, find the value of \$y\$ when \$x = $(x_val)\$.",
            answer=y_val,
            difficulty=(700, 900),
            solution=steps(
                sol("Substitute \$x = $(x_val)\$", "y = $(m)($(x_val)) + $(b)"),
                "Multiply and add",
                sol("Answer", y_val)
            ),
            time=75
        )
    end
end